recipe/conda_build_config.yaml

```diff
@@ -1,5 +1,9 @@
 python:
   - 3.9
+python_impl:
+  - cpython
+numpy:
+  - 1.19
```

Comment:
Without this `conda-smithy=3.8.2` failed with
```python
Adding in variants from internal_defaults
INFO:conda_build.variants:Adding in variants from internal_defaults
Adding in variants from /tmp/tmp600bzq31/conda_build_config.yaml
INFO:conda_build.variants:Adding in variants from /tmp/tmp600bzq31/conda_build_config.yaml
Adding in variants from /home/mbargull/code/src/conda-forge/git/python-feedstock/recipe/conda_build_config.yaml
INFO:conda_build.variants:Adding in variants from /home/mbargull/code/src/conda-forge/git/python-feedstock/recipe/conda_build_config.yaml
Traceback (most recent call last):
  File "/home/mbargull/code/conda/conda-forge/envs/conda-smithy/bin/conda-smithy", line 10, in <module>
    sys.exit(main())
  File "/home/mbargull/code/conda/conda-forge/envs/conda-smithy/lib/python3.7/site-packages/conda_smithy/cli.py", line 613, in main
    args.subcommand_func(args)
  File "/home/mbargull/code/conda/conda-forge/envs/conda-smithy/lib/python3.7/site-packages/conda_smithy/cli.py", line 419, in __call__
    self._call(args, tmpdir)
  File "/home/mbargull/code/conda/conda-forge/envs/conda-smithy/lib/python3.7/site-packages/conda_smithy/cli.py", line 430, in _call
    temporary_directory=temporary_directory,
  File "/home/mbargull/code/conda/conda-forge/envs/conda-smithy/lib/python3.7/site-packages/conda_smithy/configure_feedstock.py", line 1938, in main
    render_travis(env, config, forge_dir, return_metadata=True)
  File "/home/mbargull/code/conda/conda-forge/envs/conda-smithy/lib/python3.7/site-packages/conda_smithy/configure_feedstock.py", line 1082, in render_travis
    return_metadata=return_metadata,
  File "/home/mbargull/code/conda/conda-forge/envs/conda-smithy/lib/python3.7/site-packages/conda_smithy/configure_feedstock.py", line 602, in _render_ci_provider
    os.path.join(forge_dir, forge_config["recipe_dir"]), config=config
  File "/home/mbargull/code/conda/conda-forge/envs/conda-smithy/lib/python3.7/site-packages/conda_build/variants.py", line 525, in get_package_combined_spec
    combined_spec = combine_specs(specs, log_output=config.verbose)
  File "/home/mbargull/code/conda/conda-forge/envs/conda-smithy/lib/python3.7/site-packages/conda_build/variants.py", line 277, in combine_specs
    log_output=log_output)
  File "/home/mbargull/code/conda/conda-forge/envs/conda-smithy/lib/python3.7/site-packages/conda_build/variants.py", line 252, in _combine_spec_dictionaries
    spec_source))
ValueError: variant config in /home/mbargull/code/src/conda-forge/git/python-feedstock/recipe/conda_build_config.yaml is ambiguous because it does not fully implement all zipped keys, or specifies a subspace that is not fully implemented.
```